platform/ios/src/MGLMapView.mm

```diff
@@ -4729,7 +4736,7 @@ - (void)notifyMapChange:(mbgl::MapChange)change
                             rect = annotationView.frame;
                         }
 
-                        if ( ! CGRectIntersectsRect(rect, self.frame)) {
+                        if ( ! CGRectIntersectsRect(rect, self.frame) && self.deselectWhenOutOfBounds) {
```

Comment:
I'm happy to remove this, I just added it because I thought that @1ec5 was not fine with removing the current way how deselection is handled completely.